lib/bridge/IrcHandler.js

```diff
@@ -538,15 +538,82 @@ IrcHandler.prototype.onMode = Promise.coroutine(function*(req, server, channel,
                                                 mode, enabled, arg) {
     let privateModes = ["k", "i", "s"];
 
-    if (privateModes.indexOf(mode) === -1) {
-        return; // ignore everything but privateModes
-    }
     req.log.info(
         "onMode(%s) in %s by %s (arg=%s)",
         (enabled ? ("+" + mode) : ("-" + mode)),
         channel, by, arg
     );
 
+    if (privateModes.indexOf(mode) !== -1) {
+        req.log.info("Private mode");
+        yield this._onPrivateMode(req, server, channel, by, mode, enabled, arg);
+        return;
+    }
+
+    const matrixRooms = yield this.ircBridge.getStore().getMatrixRoomsForChannel(server, channel);
+
+    if (matrixRooms.length === 0) {
+        req.log.info("No mapped matrix rooms for IRC channel %s", channel);
+        return;
+    }
+
+    const bridgedClient = this.ircBridge.getClientPool().getBridgedClientByNick(
+        server, arg
+    );
+
+    if (!bridgedClient) {
+        req.log.info("No bridged client for %s", arg);
+        return;
+    }
+
+    if (!bridgedClient.unsafeClient) {
+        req.log.info(`Bridged client for ${arg} has no IRC client.`);
+        return;
+    }
+
+    // Bridge operator modes to power levels
+    let modeToPower = server.getModePowerMap();
+
+    if (Object.keys(modeToPower).indexOf(mode) === -1) {
```

Comment:
I'd prefer it if you did this waaaaay earlier, before we've done a DB lookup. It also makes it clearer what this code is going to handle (power level stuff).